Model/EmbeddedCallback.php

```diff
@@ -108,11 +108,12 @@ public function __construct(
      */
     public function execute()
     {
-        $request = $this->dataObjectFactory->create([
-            'data' => $this->serializer->unserialize($this->request->getContent())
-        ]);
-
         try {
+
+            $request = $this->dataObjectFactory->create([
+                'data' => $this->serializer->unserialize($this->request->getContent())
+            ]);
```

Comment:
Does this mean that usages of `$request`, like `$request->getSessionId()`, failed earlier?

Reply:
@andersem , no, I moved this part inside the try catch just to have a record in Magento log if POST body couldn't be parsed. The fix of the issue itself is adding our own Authorization model since Magento checks user type before allowing to modify cart content during API call and if user type is GUEST but we try to modify the cart which has customer_id it would throw an error 'Invalid state change requested'. USER_TYPE_INTEGRATION is allowed to modify both guest and registered customer carts. The issue happened only in specific cases.